llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -242,21 +243,15 @@ static void collectKernelModuleMap(
     KernelMapEntryScope EntryScope) {
 
   for (auto &F : M.functions()) {
-    if (F.getCallingConv() == CallingConv::SPIR_KERNEL) {
+    // Process module entry points: kernels and SYCL_EXTERNAL functions.
```

Comment:
```suggestion
    if (F.getCallingConv() != CallingConv::SPIR_KERNEL &&
        F.getCallingConv() != CallingConv::SPIR_FUNC) {
       error("Unsupported calling convention in function " + F.getName());
    }
    // Process module entry points: kernels and SYCL_EXTERNAL functions.
```

Reply:
OK, will apply.

Reply:
LowerESIMD generates `genx` intrinsics without `spif_func` calling convention. I can also add ` && F.isDeclaration()` here to make this work. Later we can fix LowerESIMD and remove ` && F.isDeclaration()` check.